examples/benchmark.py

```diff
@@ -403,63 +412,73 @@ def runOneTest(testName, options):
     # Write system info
     appendTestResult(args.outfile, system_info=system_info)
 
-if args.run_all_benchmarks:
-    # Combinatorially run all benchmarks, ignoring combinations that cannot be run
-    from openmm import OpenMMException
-    from itertools import product
-
-    if args.style == 'simple':
-        for (test, args.heavy, args.ensemble, args.platform, args.precision) in product(TESTS, HEAVY_HYDROGENS, ENSEMBLES, PLATFORMS, PRECISIONS):
+if args.platform is not None:
+    # Use specified subset of platforms
+    requested_platforms = args.platform.split(',')
+    if not set(requested_platforms).issubset(PLATFORMS):
+        parser.error(f'Available platforms: {PLATFORMS}')
+    PLATFORMS = requested_platforms
+
+if args.test is not None:
+    # Use specified subset of tests
+    requested_tests = args.test.split(',')
+    if not set(requested_tests).issubset(TESTS):
+        parser.error(f'Available tests: {TESTS}')
+    TESTS = requested_tests
+
+if args.precision is not None:
+    # Use specified subset of precisions
+    requested_precisions = args.precision.split(',')
+    if not set(requested_precisions).issubset(PRECISIONS):
+        parser.error(f'Available precisions: {PRECISIONS}')
+    PRECISIONS = requested_precisions
+
+if args.ensemble is not None:
+    # Use specified subset of ensembles
+    requested_ensembles = args.ensemble.split(',')
+    if not set(requested_ensembles).issubset(ENSEMBLES):
+        parser.error(f'Available ensembles: {ENSEMBLES}')
+    ENSEMBLES = requested_ensembles
+
+if args.bond_constraints is not None:
+    # Use specified subset of constraints
+    requested_bond_constraints = args.bond_constraints.split(',')
+    if not set(requested_bond_constraints).issubset(BOND_CONSTRAINTS):
+        parser.error(f'Available bond constraints: {BOND_CONSTRAINTS}')
+    BOND_CONSTRAINTS = requested_bond_constraints
+
+# Combinatorially run all requested benchmarks, ignoring combinations that cannot be run
+from openmm import OpenMMException
+from itertools import product
+
+if args.style == 'simple':
+    for (test, args.bond_constraints, args.ensemble, args.platform, args.precision) in product(TESTS, BOND_CONSTRAINTS, ENSEMBLES, PLATFORMS, PRECISIONS):
+        try:
+            runOneTest(test, args)
+        except OpenMMException as e:
+            pass
+elif args.style == 'rich':
+    from rich.live import Live
+    from rich.table import Table
+
+    from rich import box
+    table = Table(box=box.SIMPLE)
+    table.add_column("[red]Test", width=24)
+    table.add_column("[blue]Precision")
+    table.add_column("[blue_violet]Constraints")
+    table.add_column("[blue_violet]H mass (amu)")
+    table.add_column("[blue_violet]dt (fs)")
+    table.add_column("[blue]Platform", width=9)
+    table.add_column("[orange1]Ensemble")
```

Comment:
Rich output for "Platform" and "Ensemble" are opposite of the table headers

Reply:
Great catch---thanks!